Client/app/src/main/java/ru/hse/client/SignInActivity.kt

```diff
@@ -0,0 +1,195 @@
+                ?.newBuilder()
+                ?.addQueryParameter("login", login)
+                ?.build().toString()
+
+            val requestForGetUser: Request = Request.Builder()
+                .url(URlGetUser)
+                .build()
+
+            okHttpClient.newCall(requestForGetUser).enqueue(object : Callback {
+                override fun onFailure(call: Call, e: IOException) {
+                    Log.e("Error", e.toString() + " " + e.message)
+                    Handler(Looper.getMainLooper()).post {
+                        Toast.makeText(
+                            this@SignInActivity,
+                            "Something went wrong, try again",
+                            Toast.LENGTH_SHORT
+                        ).show()
+                    }
+                }
+
+                var badUserMessage: String = "incorrect login/password"
+
+                fun printErrorAboutBadUser() {
+                    Log.e("Info", "no such user with this login")
+                    Handler(Looper.getMainLooper()).post {
+                        Toast.makeText(
+                            this@SignInActivity,
+                            badUserMessage,
+                            Toast.LENGTH_SHORT
+                        ).show()
+                        loginLayout.error = badUserMessage
+                        passwordLayout.error = badUserMessage
+                    }
+                }
+
+                fun printOkAboutGoodUser() {
+                    Log.e("Info", "user exist")
+                    Handler(Looper.getMainLooper()).post {
+                        Toast.makeText(
+                            this@SignInActivity,
+                            "Welcome",
+                            Toast.LENGTH_SHORT
+                        ).show()
+                    }
+                }
+
+                override fun onResponse(call: Call, response: Response) {
+                    Log.i("Info", response.toString())
+                    if (response.code == 200) {
+                        val responseBody: ByteString? = response.body?.byteString()
+                        val user: EntitiesProto.UserModel =
+                            EntitiesProto.UserModel.parseFrom(responseBody?.toByteArray())
+                        if (user.hasPasswordHash()) {
+                            if (user.passwordHash == password) {
+                                printOkAboutGoodUser()
+                                // TODO: create main Activity
+                            } else {
+                                Log.e("Info", "no such login/password")
+                                printErrorAboutBadUser()
+                            }
+                        } else {
+                            Log.e("Info", "no such login/password")
+                            printErrorAboutBadUser()
+                        }
+                    } else {
+                        Log.e("Info", "no such login/password")
```

Comment:
i think better write response from server, because it can be error code 500, which means it's server error not client